eng/common/TestResources/New-TestResources.ps1

```diff
@@ -167,9 +167,11 @@ try {
         Get-ChildItem -Path $root -Filter "$_" -Recurse | ForEach-Object {
             Write-Verbose "Found template '$($_.FullName)'"
             if ($_.Extension -eq '.bicep') {
-                $templateFiles += (BuildBicepFile $_)
+                $templateFile = @{originalFilePath = $_.FullName; jsonFilePath = (BuildBicepFile $_)}
+                $templateFiles += $templateFile
             } else {
-                $templateFiles += $_.FullName
+                $templateFile = @{originalFilePath = $_.FullName; jsonFilePath = $_.FullName}
+                $templateFiles += $templateFile
             }
```

Comment:
Functionally, this is great. I wonder if there's an opportunity to do a little more refactoring here to simplify the downstream code and make it easier to avoid bugs in the future (e.g. using the wrong file path).

If you add the directory to the templateFile object here, you can avoid all the calls to `Split-Path` below and instead use `$preDeploymentScript = Join-Path $templateFile.Location 'test-resources-pre.ps1'`


```suggestion
            $templateFile = @{
                location = $_.DirectoryName
                originalFilePath = $_.FullName;
                jsonFilePath = $_.FullName
            }
            if ($_.Extension -eq '.bicep') {
                $templateFile.jsonFilePath = (BuildBicepFile $_)
            }
            $templateFiles += $templateFile
```

Just a suggestion though, feel free to ignore :)